test/hotspot/gtest/os/linux/test_cgroupSubsystem_linux.cpp

```diff
@@ -0,0 +1,140 @@
+  stringStream path;
+  path.print_raw(os::get_temp_directory());
+  path.print_raw(os::file_separator());
+  path.print("%s-test-jdk.pid%d.%s.%s", prefix, os::current_process_id(),
+             test_info->test_case_name(), test_info->name());
+  return path.as_string(true);
+}
+
+void delete_file(const char* filename) {
+  if (!file_exists(filename)) {
+    return;
+  }
+  int ret = remove(filename);
+  EXPECT_TRUE(ret == 0 || errno == ENOENT) << "failed to remove file '" << filename << "': "
+      << os::strerror(errno) << " (" << errno << ")";
+}
+
+class TestController : public CgroupController {
+public:
+  char* subsystem_path() override {
+    // The real subsystem is in /tmp/, generaed by temp_file()
+    return (char*)"/";
+  };
+};
+
+void fill_file(const char* path, const char* content) {
+  delete_file(path);
+  FILE* fp = os::fopen(path, "w");
+  if (fp == nullptr) {
+    return;
+  }
+  if (content != nullptr) {
+    fprintf(fp, "%s", content);
+  }
+  fclose(fp);
+}
+
+TEST(CgroupTest, SubSystemFileLineContentsTests) {
+  TestController my_controller{};
+  const char* test_file = temp_file("cgroups");
+  int x = 0;
+  char s[1024];
+  int err = 0;
+
+  fill_file(test_file, "1337");
+  err = subsystem_file_line_contents(&my_controller, test_file, nullptr, "%d", &x);
+  EXPECT_EQ(err, 0);
+  EXPECT_EQ(x, 1337) << "Wrong value for x";
+
+  s[0] = '\0';
+  fill_file(test_file, "1337");
+  err = subsystem_file_line_contents(&my_controller, test_file, nullptr, "%s", &s);
+  EXPECT_EQ(err, 0);
+  EXPECT_STREQ(s, "1337");
+
+  x = -1;
+  fill_file(test_file, nullptr);
+  err = subsystem_file_line_contents(&my_controller, test_file, nullptr, "%d", &x);
+  EXPECT_NE(err, 0) << "Empty file should've failed";
+  EXPECT_EQ(x, -1) << "x was altered";
+
+
+  fill_file(test_file, "foo");
+  err = subsystem_file_line_contents(&my_controller, test_file, nullptr, "%s", &s);
+  EXPECT_EQ(err, 0);
+  EXPECT_STREQ(s, "foo");
+
+  s[0] = '\0';
+  fill_file(test_file, "foo bar");
+  err = subsystem_file_line_contents(&my_controller, test_file, "foo", "%s", &s);
+  EXPECT_EQ(err, 0);
+  EXPECT_STREQ(s, "bar") << "Incorrect!";
+
+  s[0] = '\0';
+  fill_file(test_file, "faulty_start foo bar");
+  err = subsystem_file_line_contents(&my_controller, test_file, "foo", "%s", &s);
+  EXPECT_NE(err, 0) << "Key must be at start";
+
+  s[0] = '\0';
+  fill_file(test_file, "foof bar");
+  err = subsystem_file_line_contents(&my_controller, test_file, "foo", "%s", &s);
+  EXPECT_NE(err, 0) << "Key must be exact match";
+
+  s[0] = '\0';
+  fill_file(test_file, "foof bar\nfoo car");
+  err = subsystem_file_line_contents(&my_controller, test_file, "foo", "%s", &s);
+  EXPECT_EQ(err, 0);
+  EXPECT_STREQ(s, "car");
+
+  s[0] = '\0';
+  fill_file(test_file, "foo 1\nfoo car");
+  err = subsystem_file_line_contents(&my_controller, test_file, "foo", "%s", &s);
+  EXPECT_EQ(err, 0);
+  EXPECT_STREQ(s, "1");
```

Comment:
We should have a test where the separator isn't a single space, like `\t` instead. See https://github.com/jerboaa/jdk/commit/8ab1009a18a95510504f70d8a02c822083fdffd1 That's when those tests currently fail. Like I said previously it would also be good to have a test similar to how `CgroupV2Subsystem::cpu_period()` uses it. Like 
https://github.com/jerboaa/jdk/commit/4d562f08952e17f9137c0da74a3bb587c8539713

Reply:
I realize you wanted to include this as a separate RFE, but it doesn't make much sense as those additional case are (to me) the bare minimum to have in order to ensure we don't regress.